### DISCUSSION TITLE
Jira fields values in GitHub actions

### DISCUSSION THREAD

**Discussion by Srikanth419** [community/community#65881] [Actions]
### Select Topic Area

Show & Tell

### Body

I can trigger GitHub actions workflow file from JIRA board using Repository Dispatch event.
Is there any way to pass/sent JIRA ticket field values as variables to GitHub actions pipeline?
I want to use Jira field values in my GitHub actions workflow file.

**Comment by arafajs** [comment]
- When triggering the workflow, include the field values in the `client_payload` of the repository dispatch event. For example:

```
{
  "event_type": "my_event",
  "client_payload": {
    "jira_issue_key": "ABC-123", 
    "jira_assignee": "john.doe"
  }
}
```

Then in your GitHub Actions workflow YAML file, you can access them like:

```yaml
on:
  repository_dispatch:
    types: [my_event]

jobs:
  my_job:
    runs-on: ubuntu-latest
    steps:
    - name: Print variables
      run: |
        echo ${{ github.event.client_payload.jira_issue_key }}
        echo ${{ github.event.client_payload.jira_assignee }}
```

- Use the JIRA API to lookup the ticket details and save them as outputs of one job, then use those outputs as inputs in subsequent jobs. For example:

```yaml
on:
  repository_dispatch:
    types: [my_event]

jobs:

  get_ticket_details:
    runs-on: ubuntu-latest
    steps:
    - uses: atlassian/gajira-get-issue-details@v2
      id: get_issue
      with:
        issueIdOrKey: ${{ github.event.client_payload.jira_issue_key }}
    - run: |
        echo "issue_key=${{ steps.get_issue.outputs.issueKey }}" >> $GITHUB_OUTPUT 
        echo "assignee=${{ fromJSON(steps.get_issue.outputs.issue).fields.assignee.name }}" >> $GITHUB_OUTPUT

  downstream_job:
    needs: get_ticket_details 
    runs-on: ubuntu-latest
    steps:
    - run: |  
        echo ${{ needs.get_ticket_details.outputs.issue_key }}
        echo ${{ needs.get_ticket_details.outputs.assignee }}
```

- Store the JIRA ticket details as secrets in GitHub, then access them in the workflow.

So in summary, you can pass data from JIRA to GitHub Actions via the repository dispatch payload, API lookups, or GitHub secrets.

**Comment by Srikanth419** [comment]
Hello @arafajs  

Thanks for your help.

{
  "event_type": "my_event",
  "client_payload": {
    "jira_issue_key": "ABC-123", 
    "jira_assignee": "john.doe"
  }
}
Its sending same data every time. In my scenario data will change ticket to ticket.
say i will be adding new field value in another JIRA ticket in that this case also its sending same data 
.
How to overcome this.
